CI fails at `test_recursive_warm_qaoa`

### Environment

- **Qiskit Optimization version**: latest
- **Python version**: NA
- **Operating system**: NA


### What is happening?

A random error has been occurring in CI
https://github.com/Qiskit/qiskit-optimization/actions

Here is a typical log.
```
test.algorithms.test_recursive_optimization.TestRecursiveMinEigenOptimizer.test_recursive_warm_qaoa
---------------------------------------------------------------------------------------------------

Captured traceback:
~~~~~~~~~~~~~~~~~~~
    Traceback (most recent call last):

      File "/home/runner/work/qiskit-optimization/qiskit-optimization/test/optimization_test_case.py", line 47, in wrapper
    test_item(self, *args)

      File "/home/runner/work/qiskit-optimization/qiskit-optimization/test/algorithms/test_recursive_optimization.py", line 155, in test_recursive_warm_qaoa
    np.testing.assert_array_almost_equal(cplex_result.x, result.x, 4)

      File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 1046, in assert_array_almost_equal
    assert_array_compare(compare, x, y, err_msg=err_msg, verbose=verbose,

      File "/opt/hostedtoolcache/Python/3.8.12/x64/lib/python3.8/site-packages/numpy/testing/_private/utils.py", line 844, in assert_array_compare
    raise AssertionError(msg)

    AssertionError: 
Arrays are not almost equal to 4 decimals

Mismatched elements: 2 / 2 (100%)
Max absolute difference: 2.
Max relative difference: 1.
 x: array([0., 2.])
 y: array([2., 0.])
```

### How can we reproduce the issue?

Run CI

### What should happen?

No error

### Any suggestions?

I will look into the details.

--- a/test/algorithms/test_recursive_optimization.py
+++ b/test/algorithms/test_recursive_optimization.py
@@ -1,6 +1,6 @@
 # This code is part of Qiskit.
 #
-# (C) Copyright IBM 2018, 2021.
+# (C) Copyright IBM 2018, 2022.
 #
 # This code is licensed under the Apache License, Version 2.0. You may
 # obtain a copy of this license in the LICENSE.txt file in the root directory
@@ -18,7 +18,7 @@ from test import QiskitOptimizationTestCase, requires_extra_library
 import numpy as np
 
 from qiskit import BasicAer
-from qiskit.utils import algorithm_globals
+from qiskit.utils import algorithm_globals, QuantumInstance
 
 from qiskit.algorithms import NumPyMinimumEigensolver, QAOA
 
@@ -130,9 +130,15 @@ class TestRecursiveMinEigenOptimizer(QiskitOptimizationTestCase):
     @requires_extra_library
     def test_recursive_warm_qaoa(self):
         """Test the recursive optimizer with warm start qaoa."""
-        algorithm_globals.random_seed = 12345
+        seed = 1234
+        algorithm_globals.random_seed = seed
         backend = BasicAer.get_backend("statevector_simulator")
-        qaoa = QAOA(quantum_instance=backend, reps=1)
+        qaoa = QAOA(
+            quantum_instance=QuantumInstance(
+                backend=backend, seed_simulator=seed, seed_transpiler=seed
+            ),
+            reps=1,
+        )
         warm_qaoa = WarmStartQAOAOptimizer(
             pre_solver=SlsqpOptimizer(), relax_for_pre_solver=True, qaoa=qaoa
         )

